presto-main/src/test/java/com/facebook/presto/failureDetector/TestHeartbeatFailureDetector.java

```diff
@@ -88,6 +93,34 @@ public void configure(Binder binder)
         assertTrue(detector.getFailed().isEmpty());
     }
 
+    @Test
```

Comment:
thanks for the ObjectMapperProvider reference. I had idea about it but did not know where to find ;)

Added deserialization asserts to the test. However, I think that pattern match on the exception message in the json string makes a test only stronger. It helps to catch cases when exception message won't be serialized into string json properly.